zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -949,21 +1401,59 @@ mod tests {
         let account0_uuid = st.test_account().unwrap().account().id();
         let account0_id = get_account_ref(&st.wallet().db().conn, account0_uuid).unwrap();
 
-        let check = |db: &WalletDb<_, _>, account_id| {
+        // The chain height must be known in order to reserve addresses, as we store the height at
+        // which the address was considered to be exposed.
+        st.wallet_mut()
+            .db_mut()
+            .update_chain_tip(birthday.height())
+            .unwrap();
+
+        let check = |db: &WalletDb<_, _, _>, account_id| {
             eprintln!("checking {account_id:?}");
-            assert_matches!(ephemeral::first_unstored_index(&db.conn, account_id), Ok(addr_index) if addr_index == GAP_LIMIT);
-            assert_matches!(ephemeral::first_unreserved_index(&db.conn, account_id), Ok(addr_index) if addr_index == 0);
+            assert_matches!(
+                find_gap_start(&db.conn, account_id, KeyScope::Ephemeral, db.gap_limits.ephemeral()), Ok(addr_index)
+                    if addr_index == Some(NonHardenedChildIndex::ZERO)
+            );
+            //assert_matches!(ephemeral::first_unstored_index(&db.conn, account_id), Ok(addr_index) if addr_index == GAP_LIMIT);
```

Comment:
Is this assertion correct? Delete it if it isn't correct, otherwise uncomment it.